scripts/UpdatePackageVersion.swift

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/swift
+
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#if os(Linux)
+import Glibc
+#else
+import Darwin
+#endif
+import class Foundation.PropertyListDecoder
+import class Foundation.ProcessInfo
+import class Foundation.FileManager
+
+// MARK: - Main
+
+/**
+Updates the the version located at the provided file path. 
+The file stored at the path should be a plain text file that only contains the version. 
+The version should be a simple semver formatted version number e.g. MAJOR.MINOR.PATCH
+*/
+func main() {
```

Comment:
this is fine but something to think about making swift packages for these scripts.

Then we can use packages like https://github.com/apple/swift-argument-parser which makes both developer and user life easier. A lot of boilerplate can be avoided.

Reply:
We could even do code sharing with that.

Reply:
100%

I'm not tackling that rn given I'm trying to stay focused on getting the release automation wrapped up. 
I'll file an issue for this as I see this part of converting all of our scripts to swift. 

In my mind, we'll have a swift cli package (that pulls in argument parser) and will be the single entry point to all of our scripts. This will allow for code reuse (like you mentioned) as well as adding tests for our scripting logic as well.